topik/intermediaries/raw_data.py

```diff
@@ -190,6 +226,21 @@ def save(self, filename, saved_data=None):
                           "password": self.password, "doc_type": self.doc_type, "query": self.query}
         return super(ElasticSearchCorpus, self).save(filename, saved_data)
 
+    def synchronize(self, max_wait, field):
+        # TODO: change this to a more general condition for wider use?
+        # could just pass in a string condition and then 'while not eval(condition)'
+        count_not_yet_updated = -1
+        while count_not_yet_updated != 0:
```

Comment:
This looks OK for now.  TODO: can we generalize this for also the wait during read input?